apps/dash-manufacture-spc-dashboard/app.py

```diff
@@ -0,0 +1,1209 @@
+        "data": [
+            {
+                "x": x_array[:total_count],
+                "y": y_array[:total_count],
+                "mode": "lines+markers",
+                "name": col,
+                "line": {"color": "#f4d44d"},
+            },
+            ooc_trace,
+            histo_trace,
+        ]
+    }
+
+    len_figure = len(fig["data"][0]["x"])
+
+    fig["layout"] = dict(
+        margin=dict(t=40),
+        hovermode="closest",
+        uirevision=col,
+        paper_bgcolor="rgba(0,0,0,0)",
+        plot_bgcolor="rgba(0,0,0,0)",
+        legend={"font": {"color": "darkgray"}, "orientation": "h", "x": 0, "y": 1.1},
+        font={"color": "darkgray"},
+        showlegend=True,
+        xaxis={
+            "zeroline": False,
+            "showgrid": False,
+            "title": "Batch Number",
+            "showline": False,
+            "domain": [0, 0.8],
+            "titlefont": {"color": "darkgray"},
+        },
+        yaxis={
+            "title": col,
+            "showgrid": False,
+            "showline": False,
+            "zeroline": False,
+            "autorange": True,
+            "titlefont": {"color": "darkgray"},
+        },
+        annotations=[
+            {
+                "x": 0.75,
+                "y": lcl,
+                "xref": "paper",
+                "yref": "y",
+                "text": "LCL:" + str(round(lcl, 3)),
+                "showarrow": False,
+                "font": {"color": "white"},
+            },
+            {
+                "x": 0.75,
+                "y": ucl,
+                "xref": "paper",
+                "yref": "y",
+                "text": "UCL: " + str(round(ucl, 3)),
+                "showarrow": False,
+                "font": {"color": "white"},
+            },
+            {
+                "x": 0.75,
+                "y": usl,
+                "xref": "paper",
+                "yref": "y",
+                "text": "USL: " + str(round(usl, 3)),
+                "showarrow": False,
+                "font": {"color": "white"},
+            },
+            {
+                "x": 0.75,
+                "y": lsl,
+                "xref": "paper",
+                "yref": "y",
+                "text": "LSL: " + str(round(lsl, 3)),
+                "showarrow": False,
+                "font": {"color": "white"},
+            },
+            {
+                "x": 0.75,
+                "y": mean,
+                "xref": "paper",
+                "yref": "y",
+                "text": "Targeted mean: " + str(round(mean, 3)),
+                "showarrow": False,
+                "font": {"color": "white"},
+            },
+        ],
+        shapes=[
+            {
+                "type": "line",
+                "xref": "x",
+                "yref": "y",
+                "x0": 1,
+                "y0": usl,
+                "x1": len_figure + 1,
+                "y1": usl,
+                "line": {"color": "#91dfd2", "width": 1, "dash": "dashdot"},
```

Comment:
I replaced `"dashdot"` with `"dot"` to better match the mock-up. Don't forget the 3 more occurrences below if you choose to change this.

Reply:
💯